bittensor/_neuron/text/core_server/__init__.py

```diff
@@ -427,18 +432,25 @@ def synapse_check(self, synapse, hotkey):
                 return False
             
         elif synapse.synapse_type == bittensor.proto.Synapse.SynapseType.TEXT_CAUSAL_LM:
-
-            if self.metagraph.S[incoming_uid] < self.config.neuron.causallm_stake:
+            batch_size, sequence_len  =  inputs_x[0].size()
+            if (self.metagraph.S[incoming_uid] < self.config.neuron.causallm_stake) \
+                and (batch_size > self.config.neuron.max_batch_size) \
+                and (sequence_len > self.config.neuron.max_sequence_len):
```

Comment:
Should these be `or`? Otherwise I could still ddos someone with a large batch size and smaller sequence length, or vice versa.

Reply:
Fixed as well